google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineExecutor.java

```diff
@@ -45,8 +45,15 @@ public void onStart(Process process) {
   @Override
   public void cancel() {
     if (process != null) {
-      process.destroy();
-      task.onCancel();
+      // Only destroy the process and signal cancellation if the process hasn't exited
```

Comment:
I missed some details in your question - yes its called by the UI when the user hits the stop button. It would be invoked multiple times because we keep track of all deployments and don't clear out successful ones from the list. The success happens asynchronously so it would be difficult to do this - I think its safer just to attempt to cancel them all (skipping the processes that are already terminated).